lib/matplotlib/axes/_axes.py

```diff
@@ -6871,6 +6871,73 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
                           else "List[Polygon]")
             return tops, bins, cbook.silent_list(patch_type, patches)
 
+    @_preprocess_data()
+    def stairs(self, values, edges=None, *,
+               orientation='vertical', baseline=0, fill=False, **kwargs):
+        """
+        A histogram-like line or filled plot.
+
+        Parameters
+        ----------
+        values : array-like
+            An array of y-values.
+
+        edges : array-like, default: ``range(len(vals)+1)``
+            A array of x-values, with ``len(edges) == len(vals) + 1``,
+            between which the curve takes on vals values.
+
+        orientation : {'vertical', 'horizontal'}, default: 'vertical'
+
+        baseline : float or None, default: 0
+            Determines starting value of the bounding edges or when
+            ``fill=True``, position of lower edge.
+
+        fill : bool, default: False
+
+        Returns
+        -------
+        StepPatch : `matplotlib.patches.StepPatch`
+
+        Other Parameters
+        ----------------
+        **kwargs
+            `~matplotlib.patches.StepPatch` properties
+
+        """
+
+        if 'color' in kwargs:
+            _color = kwargs.pop('color')
+        else:
+            _color = self._get_lines.get_next_color()
+        if fill:
+            kwargs.setdefault('edgecolor', 'none')
+            kwargs.setdefault('facecolor', _color)
+        else:
+            kwargs.setdefault('edgecolor', _color)
+
+        if edges is None:
+            edges = np.arange(len(values) + 1)
+
+        self._process_unit_info(xdata=edges, ydata=values, kwargs=kwargs)
+        edges = self.convert_xunits(edges)
+        values = self.convert_yunits(values)
+
+        patch = mpatches.StepPatch(values,
+                                   edges,
+                                   baseline=baseline,
+                                   orientation=orientation,
+                                   fill=fill,
+                                   **kwargs)
+        self.add_patch(patch)
+        if baseline is None:
+            baseline = 0
+        if orientation == 'vertical':
+            patch.sticky_edges.y.append(baseline)
+        else:
+            patch.sticky_edges.x.append(baseline)
```

Comment:
just looking from the peanut gallery...
1) do you actually want a sticky at y=0 if baseline is None?
2) adding the sticky *could* be done as part of the StepPatch implementation (you could even try to keep it in sync in `set_baseline`, although it's not necessarily clear what to do if the user manually changed the value in the meantime, so... perhaps not).  I realize other classes don't do that, although for many of them that's because they're too general for the class to really know what stickies to use (e.g. a Rectangle does not know it's being used as part of a histogram).

Reply:
1 is intended. I think the  implied consensus was that the naming would be general but some histogram like assumptions could be made in the plotting function. That's why sticky is in stairs and not in StepPatch. I suppose that means the update of StepPatch is not nicely behaved wrt baseline sticky, but then again neither are lims